### PR TITLE
ztp: CNF-14638: Backport CLO6 changes to release-4.16

### PR DESCRIPTION
**Pull request by fontivan**
Includes 5 PRs that were merged into main

- https://github.com/openshift-kni/cnf-features-deploy/pull/2028
- https://github.com/openshift-kni/cnf-features-deploy/pull/2042
- https://github.com/openshift-kni/cnf-features-deploy/pull/2046
- https://github.com/openshift-kni/cnf-features-deploy/pull/2049
- https://github.com/openshift-kni/cnf-features-deploy/pull/2053

Dropped ztp/kube-compare changes from the originals PRs as kube-compare is not present in release-4.16

Also includes tooling changes from #1925 to update the policy generator plugin from 1.12.4 to latest since it is required for the new CRs

### CODE REVIEW

**Comment by fontivan** [comment]
Undrafting PR to allow CI to run though PR still held pending final approvals

File: ztp/gitops-subscriptions/argocd/example/acmpolicygenerator/README.md [hunk 1]
```diff
@@ -96,4 +111,37 @@ The DU profile ACM PolicyGenerator examples can be used with the same [policies-
...
+The policy which removes the CLO 5.y API artifacts must account for two
+scenarios
+ - upgraded clusters where the 5.y CRDs and artifacts exist and must be removed
+ - newly deployed clusters where the 5.y CRDs never existed and the types are thus unknown
+
+To avoid having two separate policies, one for upgrade and one for newly
+deployed clusters, the `acm-group-du-clo5-cleanup` policy includes
+`ClusterLogging5Cleanup.yaml` which is not a true "source CR". This file is an
+ACM Policy `object-template-raw` which enables us to query for existence of the
+CRD and, iff it exists, remove the old API CR and the CRD. This leverages the
```
Comment (sabbir-47) [inline ztp/gitops-subscriptions/argocd/example/acmpolicygenerator/README.md:135]:
```suggestion
CRD and, if it exists, remove the old API CR and the CRD. This leverages the
```

**Comment by fontivan** [comment]
@ffromani Can you review/approve this due to the tooling changes

**Comment by fontivan** [comment]
@imiller0 Can you review/approve this due to the tooling changes

**Comment by imiller0** [comment]
Backport of CLO lgtm. The update to set golang version to 1.21 aligns with other core openshift operators on 4.16 branches.

**Comment by sabbir-47** [comment]
@imiller0 can we get an /approve from you? Thanks

**Pull request merged**